cmd/kube-proxy/app/server.go

```diff
@@ -576,25 +578,29 @@ func createClients(config componentbaseconfig.ClientConnectionConfiguration, mas
 	return client, eventClient.CoreV1(), nil
 }
 
-func serveHealthz(hz healthcheck.ProxierHealthUpdater) {
+func serveHealthz(hz healthcheck.ProxierHealthUpdater, hardFail bool, errCh chan error) {
 	if hz == nil {
 		return
 	}
 
 	fn := func() {
 		err := hz.Run()
 		if err != nil {
-			// For historical reasons we do not abort on errors here.  We may
-			// change that in the future.
-			klog.Errorf("healthz server failed: %v", err)
+			klog.Errorf("starting healthz server failed: %v", err)
```

Comment:
It's not an error starting, it's an error.  If the HTTP server starts and later fails, this path is hit.  The original error string is correct